modules/fedimint-mint-server/src/lib.rs

```diff
@@ -363,15 +361,143 @@ impl ServerModule for Mint {
         consensus_items: Vec<(PeerId, MintConsensusItem)>,
         _consensus_peers: &BTreeSet<PeerId>,
     ) -> Vec<PeerId> {
-        for (peer, consensus_item) in consensus_items {
-            self.process_partial_signature(
-                dbtx,
-                peer,
-                consensus_item.out_point,
-                consensus_item.signatures,
+        for (peer_id, consensus_item) in consensus_items {
+            let out_point = consensus_item.out_point;
+            let signatures = consensus_item.signatures;
+
+            // check if we already obtained the blinded threshold signature for this
+            // out_point
+            if dbtx.get_value(&OutputOutcomeKey(out_point)).await.is_some() {
+                continue;
+            }
+
+            // check if we already have a valid signature share for this peer
+            if dbtx
+                .get_value(&ReceivedPartialSignatureKey(out_point, peer_id))
+                .await
+                .is_some()
+            {
+                continue;
```

Comment:
nit: we might want to log this?

Reply:
if all nodes are correct and contribute we would log this f times for every created blind signature though.

Reply:
Done